ci/README.md

```diff
@@ -0,0 +1,141 @@
+# Installing Spinnaker for GCP on a Continous Integration Server
+
+Spinnaker for GCP supports installation through a continous integration Server. A CI server can be used to conduct the initial installation and to apply updates when the Spinnaker for GCP properties file changes. Solutions for Google Cloud Build and Jenkins are detailed below.
+
+## Google Cloud Build
+
+### Shared VPC
+
+Please note that using Cloud Build to install Spinnaker for GCP with a Shared VPC host project is currently unsupported. For Shared VPC support, conduct the [setup](https://cloud.google.com/docs/ci-cd/spinnaker/spinnaker-for-gcp) in Cloud Shell.
+
+### Service account
+
+Google Cloud Build has a default service account that can be found in the settings page of the Cloud Build UI. The Cloud Build service account must be configured with the following roles to install Spinnaker for GCP:
+
+- Cloud Functions Developer - roles/cloudfunctions.developer
+- Compute Network Viewer - roles/compute.networkViewer
+- Kubernetes Engine Admin - roles/container.developer 
+- Create Service Accounts - roles/iam.serviceAccountCreator
+- Pub/Sub Editor - roles/pubsub.editor
+- Cloud Memorystore Redis Admin - roles/redis.admin
+- Service Usage Admin - roles/serviceusage.serviceUsageAdmin
+- Source Repository Administrator - roles/source.admin
+- Storage Admin - roles/storage.admin
+- Project IAM Admin - roles/resourcemanager.projectIamAdmin
```

Comment:
A number of admin-level permissions here - do we know to what extent they're all needed?

Reply:
Sure, I started with the least privilege and worked my way up as the setup ran into obstacles:

- Kubernetes Engine Admin: For creating a new cluster and for managing the API objects
- Cloud Memorystore Redis Admin: For creating a redis instance
- Service Usage Admin: For enabling required service apis
- Source Repository Administrator: For creating a new repo
- Storage Admin: For creating a new bucket
- Project IAM Admin: For getting the IAM roles on the service account executing the script and determining if the install can proceed